optica.rb

```diff
@@ -8,6 +8,15 @@ class Optica < Sinatra::Base
   end
 
   get '/' do
+    return get_nodes(request, true)
+  end
+
+  # endpoint for fab usage
+  get '/roles' do
+    return get_nodes(request, false)
+  end
+
+  def get_nodes(request, is_full_node)
```

Comment:
good idea. Do you think we should add support for the clients to pass fields_to_include themselves? I will refactor and merge, we can discuss the idea of adding support for parameterized later.